features/networking/egress-ip.feature

```diff
@@ -232,3 +232,127 @@ Feature: Egress IP related features
     Then the step should succeed
     And the output should not contain "reg0=0x"
     """
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-19973
+  @admin
+  @destructive
+  Scenario: The egressIP should still work fine after the node or network service restarted
+    Given the valid egress IP is added to the node
+    And I have a project
+    And I have a pod-for-ping in the project
+
+    # add the egress ip to the project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ip %>"]} |
+
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | ifconfig.me |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+
+    # Reboot the node which patched egressIP
+    Given the host is rebooted and I wait it up to 600 seconds to become available
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | ifconfig.me |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-15998
+  @admin
+  Scenario Outline: Invalid egressIP should not be acceptable
+    Given I select a random node's host
+    When I run the :patch admin command with:
+      | resource      | hostsubnet                     |
+      | resource_name | <%= node.name %>               |
+      | p             | {"egressIPs": <invalid_ip> }   |
+      | type          | merge                          |
+    Then the step should fail
+    And the output should contain "Invalid JSON Patch"
+
+    Examples:
+      | invalid_ip              |
+      | fe80::5054:ff:fedd:3698 |
+      | a.b.c.d                 |
+      | 10.10.10.-1             |
+      | 10.0.0.1/64             |
+      | 10.1.1/24               |
+      | A008696                 |
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-25694
+  @admin
+  @destructive
+  Scenario: Random outages with egressIP
+    Given I store the masters in the :nodes clipboard
+    And the valid egress IP is added to the "<%= cb.nodes[0].name %>" node
```

Comment:
Does this step is to add egressIP to master node? I assume the egressIP is needed to add to work nodes

Reply:
Yes, it is intended to add the egressIP to the master node. That insure the test pod can be on different node from the egressIP node.

Reply:
> Yes, it is intended to add the egressIP to the master node. That insure the test pod can be on different node from the egressIP node.

Should egressIP only be configured on worker node not master node? I do not think egressIP on the master is a real user case

Reply:
I can change it, but I don't think there is difference to configure it on master node or worker node. And generally vsphere resource is limited, from my personal testing, I generally deploy one worker node there.

Reply:
For real user case, using worker nodes make sense because the lots of will be redirected to the egressIP nodes. But for our testing purpose, it is just one simple curl command. I think it is okay to configure it on master node.

Reply:
For the testing results, you are right we may not see the difference, but master nodes and worker nodes take different roles in the cluster, same as the testing for egressrouter which should be deployed in the worker nodes not master nodes, even you can do this way.

I can image if this test case fail in the future, the dev will ask QE to reproduce the problem by changing to deploy the egreesIP in the worker nodes. In order to avoid wasting time in the future, I would suggest to update the step to use worker node now. WDYT?

Reply:
@weliang1 Fixed it, thanks!